### ISSUE TITLE
Update() shouldn't load existing entity

### ISSUE THREAD

**Issue by Eagle3386** [CypressNorth/.NET-EF6-GenericRepository#6]
Since EF5+ [automatically attaches](https://msdn.microsoft.com/en-us/library/gg696238.aspx) an entity whenever it is marked as modified, your code:

``` c#
TObject existing = await _context.Set<TObject>().FindAsync(key);
if (existing != null)
{
  _context.Entry(existing).CurrentValues.SetValues(updated);
  await _context.SaveChangesAsync();
}
```

should be updated to:

``` c#
_context.Entry(updated).State = EntityState.Modified;
await _context.SaveChangesAsync();
```

This gives three huge advantages:
a) only one database trip
b) less resource load, because no need for loading the original entity
c) no need for specifying a key, hence no concrete type dependence and a true generic method

Please see [this StackOverflow comment](http://stackoverflow.com/questions/15336248/entity-framework-5-updating-a-record#comment30861474_15339512) for further details.

**Comment by mombrea** [comment]
Thanks @Eagle3386 , as soon as I have a chance to test I will update the repo.

**Comment by Eagle3386** [comment]
You're welcome. I even created a PR, so you don't have to write the code on your own.

**Comment by Eagle3386** [comment]
Note: GitHub kind of "updated" my pull request: I intended to provide you with another PR which improved formatting, changed a couple of methods in order to return IQueryable in favor over ICollection so that anybody using this class can further add constraints, do aggregates like Sum() and most importantly keep actual data loading out of the class as much as possible.

However, GitHub didn't let me to do so or I was too stupid to click the right buttons. Anyway, as you can see within the PR, there are atomic commits, so you can do cherry-picking if you don't like these changes.

**Comment by mombrea** [comment]
Please update the pull request to fix issues. 
https://github.com/CypressNorth/.NET-EF6-GenericRepository/pull/7

**Comment by Eagle3386** [comment]
Fixed by PR #8.
